pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1376,8 +1380,11 @@ private CompletableFuture<Void> delete(boolean failIfHasSubscriptions,
             }
 
             fenceTopicToCloseOrDelete(); // Avoid clients reconnections while deleting
+            // Mark the progress of close to prevent close calling concurrently.
+            this.closeWithoutWaitingClientFuture = new CompletableFuture<>();
+            this.fullyCloseFuture = new CompletableFuture<>();
```

Comment:
Could these fields `closeWithoutWaitingClientFuture` and `fullyCloseFuture` be final fields which have `CompletableFuture` instances that are created when PersistentTopic is constructed?
The main motivation would be to ensure that thread safety doesn't break later if these fields are after some later changes accessed without considering thread safety that is currently provided by the lock.

Reply:
`Topic.close` and `Topic.delete` may be called more than once. See `unfenceTopicToResume()`, so both futures can not be marked as `final`.